src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts

```diff
@@ -106,9 +120,13 @@ export default function useClientSideSmartOrderRouterTrade<TTradeType extends Tr
 
     // Returns the last trade state while syncing/loading to avoid jank from clearing the last trade while loading.
     if (!error) {
+      if (isStale) {
+        return { state: TradeState.LOADING, trade: undefined }
+      }
```

Comment:
This should be linked with isDebouncing with an `else if`, as they're all mutually exclusive.